### PR TITLE
📖 Add missing clusterctl commands in CAPI book

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

This PR does the following:
1) Adds missing clusterctl commands in the CAPI book, and
2) Sorts the commands in alphabetic order (like clusterctl). 

Reference `clusterctl` output
```shell
Available Commands:
  alpha       Commands for features in alpha.
  backup      Backup Cluster API objects and all dependencies from a management cluster.
  completion  Output shell completion code for the specified shell (bash or zsh)
  config      Display clusterctl configuration.
  delete      Delete one or more providers from the management cluster.
  describe    Describe workload clusters.
  generate    Generate yaml using clusterctl yaml processor.
  get         Get info from a management or a workload cluster
  help        Help about any command
  init        Initialize a management cluster.
  move        Move Cluster API objects and all dependencies between management clusters.
  restore     Restore Cluster API objects from file by glob. Object files are searched in config directory
  upgrade     Upgrade core and provider components in a management cluster.
  version     Print clusterctl version.
```

### CODE REVIEW

File: docs/book/src/clusterctl/commands/help.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# clusterctl help
+
+Help provides help for any command in the application.
+Simply type clusterctl help [path to command] for full details.
```
Comment (ramineni) [inline docs/book/src/clusterctl/commands/help.md:4]: just [command] would be suffice here, path to command would be confusing?

Comment (chrischdi) [inline docs/book/src/clusterctl/commands/help.md:4]:
```suggestion
Simply type `clusterctl help [command]` for full details.
```

Comment (oscr) [reply docs/book/src/clusterctl/commands/help.md]: Fixed! Thank you for the suggestion

**Review by killianmuldoon — COMMENTED** [review]
Good idea to fully document these - for organisation purposes maybe have all the new ones (where the command description is just one paragraph) link to a single "more-commands.md) and  the full commands page can link to each of their sections in turn.

File: docs/book/src/clusterctl/commands/config-repositories.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# clusterctl config repositories
+
+Display the list of providers and their repository configurations.
+
+clusterctl ships with a list of known providers; if necessary, edit
+$HOME/.cluster-api/clusterctl.yaml file to add new provider or to customize existing ones.
```
Comment (killianmuldoon) [inline docs/book/src/clusterctl/commands/config-repositories.md:6]:
```suggestion
$HOME/.cluster-api/clusterctl.yaml file to add a new provider or to customize existing ones.
```

Comment (oscr) [reply docs/book/src/clusterctl/commands/config-repositories.md]: Thanks! Fixed

**Comment by oscr** [comment]
> Good idea to fully document these - for organisation purposes maybe have all the new ones (where the command description is just one paragraph) link to a single "more-commands.md) and the full commands page can link to each of their sections in turn.

@killianmuldoon Great suggestion. Fixed!

**Review by killianmuldoon — COMMENTED** [review]
Could you add the new page in the Summary.md too?

**Comment by oscr** [comment]
@killianmuldoon Thanks for the suggestion. Fixed!

**Review by killianmuldoon — APPROVED** [review]
Awesome!

/lgtm

**Comment by oscr** [comment]
> One tiny nit (completely my fault too! face_exhaling
> 
> Could you change 'more commands' to 'additional commands' in the filename and the Summary?
> 
> I think it reads better.

No problem. Done!

**Comment by sbueringer** [comment]
Thx!

/lgtm
/cherry-pick release-1.2

**Review by ykakarap — COMMENTED** [review]
Mostly LGTM, just one comment.

Can you also please add a section for `clusterctl generate provider`?

**Comment by oscr** [comment]
> Mostly LGTM, just one comment.
> 
> Can you also please add a section for `clusterctl generate provider`?

@ykakarap No problem at all. Fixed!

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Pull request merged**